benchmarks/sidekiq_test.rb

```diff
@@ -0,0 +1,134 @@
+# rubocop:disable all
+ENV['RAILS_ENV'] = 'production'
+
+require 'bundler/setup'
+require 'rails/all'
+Bundler.require(*Rails.groups)
+
+module SampleApp
+  class Application < Rails::Application; end
+end
+
+module OverrideConfiguration
+  def paths
+    super.tap { |path| path.add 'config/database', with: 'benchmarks/postgres_database.yml' }
+  end
+end
+
+Rails::Application::Configuration.prepend(OverrideConfiguration)
+
+Rails.application.configure do
+  config.cache_classes = true
+  config.eager_load = true
+  config.active_job.queue_adapter = :sidekiq
+end
+
+ActiveRecord::Base.configurations = Rails.application.config.database_configuration
+Rails.application.initialize!
+
+ActiveRecord::Schema.define do
+  drop_table(:samples) if connection.table_exists?(:samples)
+
+  create_table :samples do |t|
+    t.string :name
+    t.timestamps
+  end
+end
+
+class Sample < ActiveRecord::Base; end
+
+require 'sidekiq/launcher'
+require 'sidekiq/cli'
+require 'concurrent/atomic/atomic_fixnum'
+
+Sidekiq.configure_server do |config|
+  redis_conn = proc { Redis.new(host: 'localhost', port: 6379) }
+  config.redis = ConnectionPool.new(size: 27, timeout: 3, &redis_conn)
+end
+
+options = Sidekiq.options.tap do |options|
+  options[:tag] = 'test'
+  options[:queues] << 'default'
+  options[:concurrency] = 20
+  options[:timeout] = 2
+end
+
+class Worker
+  class << self
+    attr_reader :iterations, :conditional_variable
+  end
+
+  @iterations = Concurrent::AtomicFixnum.new(0)
+  @conditional_variable = ConditionVariable.new
+
+  include Sidekiq::Worker
+  def perform(iter, max_iterations)
+    self.class.iterations.increment
+    self.class.conditional_variable.broadcast if self.class.iterations.value > max_iterations
+
+    Sample.create!(name: iter.to_s).save
+
+    100.times do
+      Sample.last.name
+    end
+
+    Sample.last(100).to_a
+  end
+end
+
+if Datadog.respond_to?(:configure)
+  Datadog.configure do |d|
+    d.use :rails,
+          enabled: true,
+          auto_instrument_redis: true,
```

Comment:
This is old syntax: does not work. Should be updated to new configuration.

Reply:
in new configuration this is Noop - as its enabled by default. I've remove these lines not to confuse anyone. It was hover useful when benchmarking older versions